### PR TITLE
refactor(ui): normalization of tasks

### PR DESCRIPTION
**Pull request by 121watts**
RFC normalization epic: https://github.com/influxdata/idpe/issues/5591

### CODE REVIEW

File: ui/src/schemas/index.ts [hunk 1]
```diff
@@ -69,3 +84,10 @@ export const arrayOfTelegrafs = [telegraf]
 
 export const scraper = new schema.Entity(ResourceType.Scrapers)
 export const arrayOfScrapers = [scraper]
+
+export const addLabels = <R extends {labels?: Label[]}>(resource: R): R => {
```
Comment (ebb-tide) [inline ui/src/schemas/index.ts]: ooh- do resource types come out of this function with required labels properties? I wonder if the last `R` on this line should be `R & {labels: Label[]}` instead?

File: ui/src/types/tasks.ts [hunk 2]
```diff
@@ -1,6 +1,36 @@
 import {Task as ITask} from 'src/client'
-import {Label} from 'src/types'
+import {Label, NormalizedState, Run, RemoteDataState, LogEvent} from 'src/types'
 
 export interface Task extends ITask {
   labels?: Label[]
```
Comment (ebb-tide) [inline ui/src/types/tasks.ts]: I feel like this can be required since we are adding defaults?

File: ui/src/utils/api.ts [hunk 3]
```diff
@@ -6,25 +6,25 @@ import {getAPIBasepath} from 'src/utils/basepath'
...
 
-  if (message === '') {
-    message = get(e, 'message', '')
+  if (!message) {
+    message = get(e, 'message')
   }
 
-  if (message === '') {
+  if (!message) {
     message = 'unknown error'
```
Comment (ebb-tide) [inline ui/src/utils/api.ts]: ah thank you for this!

File: ui/src/schemas/index.ts (hunk 1 above)
Comment (ebb-tide) [reply ui/src/schemas/index.ts]: Separately- should we not be adding labels to buckets?

Comment (ebb-tide) [reply ui/src/schemas/index.ts]: There's also a version of this in `src/labels/utils/index.ts` should this replace it?

Comment (121watts) [reply ui/src/schemas/index.ts]: I'll be doing a whole labels overhaul during the `Label` normalization epic!

File: ui/src/types/tasks.ts (hunk 2 above)
Comment (121watts) [reply ui/src/types/tasks.ts]: Good catch!  I wanted to do the bare minimum until I normalize the `Labels` resource.  All of this is going to change when labels within a resource are just a list of label ids.

File: ui/src/utils/api.ts (hunk 3 above)
Comment (121watts) [reply ui/src/utils/api.ts]: be the change

**Pull request merged**